### ISSUE TITLE
Command sequence in blocky installation

### ISSUE THREAD

**Issue by MaKeG0** [xaizone/xaizone.eu#1]
Hello and thanks for your blocky guide, i found 1 problem and i have 1 question for you.

- Problem:
the starting setup that you wrote have this:
  ```
  /interface/bridge/port add bridge=containers interface=veth1
  /interface/veth/add name=veth1 address=172.17.0.2/24 gateway=172.17.0.1
  ```
  but i've got an error and also checking the documentation those 2 commands need to reverse, or the second needs to go on top of all commands.

- Now the question as i could not find another way to contact you:
i'm pretty new to RouteOS and your guide is my first try with contaners inside of it.

  My firewall didn't like it very much 
  (i've followed the "Building Advanced Firewall" from Mikrotik documentation. [https://help.mikrotik.com/docs/display/ROS/Building+Advanced+Firewall](url))
  
  Luckly the rules that were blocking had log active and were those:
  
  1. Raw prerouting
    `/ip firewall raw add action=drop chain=prerouting comment="defconf: drop the rest"`
    that i've fixed with this:
    `/ip firewall raw add action=accept chain=prerouting comment="defconf: accept everything else from containers" in-interface=containers place-before=<number of the rule above>`
  
  1. Drop invalid
   `/ip firewall filter add action=drop chain=forward comment="defconf: drop invalid" connection-state=invalid`
   that i've fixed with this:
   `/ip firewall filter add chain=forward action=accept src-address=<LANIP> dst-address=172.17.0.2 log=yes log-prefix="allowed_to_blocky" place-before=<number of the rule above> comment="Allow forward to blocky"`
    where i've added the log just for checking but can be turned off after.
  
  This said now i get no more error log messages and connection is working fine.
  I'm just very confused if blocky is actually working as i can see the container as active, but the test you gave on your blog [https://d3ward.github.io/toolz/adblock.html](url) if i run it from chrome fails miserably. 
  (if i run it from brave it's almost 100% green but i'm not testing the browser)

Hope you might help and thanks again.

Kevin

**Comment by xaizone** [comment]
Hey,

thanks for reading my blog.

The problem seems to be related to how RouterOS handles exports; all of the commands that are in my guide are from a simple `export` command and I didn't bother to reproduce this setup on a fresh RouterOS install. Thanks for the note, I will update the post.

Regarding your question. As long as you DST-NAT all port 53 traffic from your LAN bridge to the container's IP port 6969 (from my guide), it **should** just work, or at least it did in my case, since I didn't bother with any firewall rules.

Since the time of this guide I switched to a Pihole + Unbound LXC on my new home server due to performance issues on the Mikrotik. I never managed to find what was causing this, I believe it was something related to DoT. I was getting an average response time of around 150ms, which was unbearably slow for me.

I just ran the test I mention on the latest Google Chrome version (116.0.5845.111), works just fine. It's a different story with Brave, they do a lot more under the hood regarding ads and it's all enabled by default from what I remember.
